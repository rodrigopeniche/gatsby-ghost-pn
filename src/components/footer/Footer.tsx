import * as React from "react"
import FooterContainer from "./FooterContainer"
import WebsiteAdress from "./WebsiteAdress"
import ContactsList from "./ContactsList"

const Footer = () => {
    return (
        <FooterContainer>
            <WebsiteAdress />
            <ContactsList />
        </FooterContainer>
    )
}

export default Footer
