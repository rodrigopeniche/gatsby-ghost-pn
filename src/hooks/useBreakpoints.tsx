import * as React from "react"
import { useState, useEffect, createContext, useContext } from "react"

const defaultValue = {}

const BreakpointContext = createContext(defaultValue)

const BreakpointProvider = ({
    children,
    queries,
}: {
    children: React.ReactNode | React.ReactNode[]
    queries: { [key: string]: string }
}) => {
    const [queryMatch, setQueryMatch] = useState({})

    useEffect(() => {
        const mediaQueryLists: { [key: string]: any } = {}
        const keys = Object.keys(queries)
        let isAttached = false

        const handleQueryListener = () => {
            const updatedMatches = keys.reduce(
                (acc: { [key: string]: any }, media) => {
                    acc[media] = !!(
                        mediaQueryLists[media] && mediaQueryLists[media].matches
                    )
                    return acc
                },
                {}
            )
            setQueryMatch(updatedMatches)
        }

        if (window && window.matchMedia) {
            const matches: { [key: string]: any } = {}
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media] = window.matchMedia(queries[media])
                    matches[media] = mediaQueryLists[media].matches
                } else {
                    matches[media] = false
                }
            })
            setQueryMatch(matches)
            isAttached = true
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media].addListener(handleQueryListener)
                }
            })
        }

        return () => {
            if (isAttached) {
                keys.forEach(media => {
                    if (typeof queries[media] === "string") {
                        mediaQueryLists[media].removeListener(
                            handleQueryListener
                        )
                    }
                })
            }
        }
    }, [queries])

    return (
        <BreakpointContext.Provider value={queryMatch}>
            {children}
        </BreakpointContext.Provider>
    )
}

function useBreakpoint(): { [key: string]: boolean } {
    const context = useContext(BreakpointContext)
    if (context === defaultValue) {
        throw new Error("useBreakpoint must be used within BreakpointProvider")
    }
    return context
}
export { useBreakpoint, BreakpointProvider }
