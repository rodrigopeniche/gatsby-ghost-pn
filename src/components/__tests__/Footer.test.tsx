import * as React from "react"
import Footer from "../footer/Footer"
import * as renderer from "react-test-renderer"
import { BreakpointProvider } from "../../hooks/useBreakpoints"

describe("footer", () => {
    it("renders correctly", () => {
        const tree = renderer.create(
            <BreakpointProvider queries={{}}>
                <Footer />
            </BreakpointProvider>
        )
        expect(tree.toJSON()).toMatchSnapshot()
    })
})
