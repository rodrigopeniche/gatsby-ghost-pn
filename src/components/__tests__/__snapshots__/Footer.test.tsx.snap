// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders correctly 1`] = `
<div
  style={
    Object {
      "background": "rgb(218,221,225)",
    }
  }
>
  <div
    style={
      Object {
        "background": "#EBEEF2",
        "boxShadow": "inset 0px 4px 20px rgba(0, 0, 0, 0.25)",
        "display": "flex",
        "flexDirection": "column",
        "height": "100%",
        "margin": "0 auto",
        "width": "960px",
      }
    }
  >
    <h1
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Georgia",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "18px",
          "margin": "82px auto 30px auto",
        }
      }
    >
      WWW.PHONENUMBERS.ORG
    </h1>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "margin": "0 auto 70px auto",
        }
      }
    >
      <p
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": "18px",
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        Home
      </p>
      <p
        style={
          Object {
            "display": "none",
          }
        }
      >
        |
      </p>
      <p
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": "18px",
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        FAQ
      </p>
      <p
        style={
          Object {
            "display": "none",
          }
        }
      >
        |
      </p>
      <p
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": "18px",
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        Help
      </p>
      <p
        style={
          Object {
            "display": "none",
          }
        }
      >
        |
      </p>
      <p
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": "18px",
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        Privacy
      </p>
      <p
        style={
          Object {
            "display": "none",
          }
        }
      >
        |
      </p>
      <p
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": "18px",
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        Do Not Sell My Info
      </p>
      <p
        style={
          Object {
            "display": "none",
          }
        }
      >
        |
      </p>
      <p
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": "18px",
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        Terms of Use
      </p>
    </div>
  </div>
</div>
`;
